public/app/core/components/TagFilter/TagFilter.tsx

```diff
@@ -40,7 +40,7 @@ const filterOption = (option: any, searchQuery: string) => {
 
 export const TagFilter: FC<Props> = ({
   allowCustomValue = false,
-  formatCreateLabel,
+  formatCreateLabel = (input: string) => `Use custom value: ${input}`,
```

Comment:
```suggestion
  formatCreateLabel = (input: string) => (<span>Add tag</span> <TagBadge label={label} removeIcon={false} />),
```

Maybe something like that is better? but then it wont solve the bug. 

Maybe in the TagOption component we check if label is not string we just return label (instead of the markup )

Reply:
> Maybe in the TagOption component we check if label is not string we just return label (instead of the markup )

@torkelo that would look like:
| before | after |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/20999846/174813026-7d418e43-b956-431e-bb8d-233d3b2acb6e.png) | ![image](https://user-images.githubusercontent.com/20999846/174812294-a906c70e-753f-43b1-9600-2e9923621266.png) |

we lose the tag colour on the creatable option... i don't mind too much tbh, think it looks fairly clean still and the code is nicer that way. @grafana/user-essentials any thoughts? 🤔

Reply:
yea, that works pretty well (after)

Reply:
🤔 just to double-check,  we had this "use custom value" because we support the usage of template variables, as the documentation explains:

> In Grafana v5.3+ it’s possible to use template variables in the tag query. So if you have a dashboard showing stats for different services and a template variable that dictates which services to show, you can now use the same template variable in your annotation query to only show annotations for those services.

Does this mean we are removing that?

https://user-images.githubusercontent.com/239999/175044605-73398f77-b690-472f-a089-75a293ad77a7.mp4

Reply:
@axelavargas yes, we support custom values so you can use template variables

> Does this mean we are removing that?

no - this PR means the styling has changed (see the `Hit enter to add` bit). the bug you're seeing where it doesn't actually get added is https://github.com/grafana/grafana/issues/51171, which was caused by refactoring the `TagFilter` component 😞